spacemesh/v2alpha1/tx.proto

```diff
@@ -59,4 +59,106 @@ message TransactionV1 {
 
 message Nonce {
   uint64 counter = 1;
+}
+
+// TransactionState is the "journey" of a tx from mempool to block inclusion to
+// mesh to STF processing. To know whether or not the tx actually succeeded,
+// and its side effects, check the Receipt in the GlobalStateService.
+message TransactionState {
+  bytes id = 1;
+  enum TransactionState {
+    TRANSACTION_STATE_UNSPECIFIED = 0; // default state
+    TRANSACTION_STATE_REJECTED = 1; // rejected from mempool due to, e.g., invalid syntax
+    TRANSACTION_STATE_INSUFFICIENT_FUNDS = 2; // rejected from mempool by funds check
+    TRANSACTION_STATE_CONFLICTING = 3; // rejected from mempool due to conflicting counter
+    TRANSACTION_STATE_MEMPOOL = 4; // in mempool but not on the mesh yet
+    TRANSACTION_STATE_MESH = 5; // submitted to the mesh
+    TRANSACTION_STATE_PROCESSED = 6; // processed by STF; check Receipt for success or failure
+  }
+  TransactionState state = 2;
+}
+
+message TransactionResult {
+  enum Status {
+    SUCCESS = 0;
+    FAILURE = 1;
+    INVALID = 2;
+  }
+
+  Transaction tx = 1;
+  Status status = 2;
+  string message = 3;
+  uint64 gas_consumed = 4; // in units of gas
+  uint64 fee = 5; // in smidge
+  bytes block = 6;
+  uint32 layer = 7;
+  repeated string touched_addresses = 8;
+}
+
+message TransactionStreamRequest {
+  uint32 start_layer = 1;
```

Comment:
are `start_layer` and/or `end_layer` optional? would be good to add a comment here to that effect (or can we indicate in the protobuf definition that they're optional?)

Reply:
Added comments